src/charm.py

```diff
@@ -86,11 +99,28 @@ def __init__(self, *args) -> None:
         self._peer_data = relations.secrets.RelationSecrets(self, self._PEER_RELATION_NAME)
 
         self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.config_changed, self.reconcile)
-        self.framework.observe(
-            self.on[rock.CONTAINER_NAME].pebble_ready, self._on_workload_container_pebble_ready
-        )
-        self.framework.observe(self.on.stop, self._on_stop)
+        try:
+            self.refresh = charm_refresh.Kubernetes(
+                _KubernetesRouterRefresh(
+                    workload_name="Router",
+                    charm_name="mysql-router-k8s",
+                    oci_resource_name="mysql-router-image",
+                )
+            )
+        except charm_refresh.UnitTearingDown:
+            # MySQL server charm will clean up users & router metadata when the MySQL Router app or
+            # unit(s) tear down
+            self.unit.status = ops.MaintenanceStatus("Tearing down")
+            self._reconcile_allowed = False
+        except charm_refresh.KubernetesJujuAppNotTrusted:
```

Comment:
If charm already deployed, can we just assumed trust is given or can it be revoked? Does this ever gets thrown